spec/syncSpec.js

```diff
@@ -285,26 +320,24 @@ function doCreate(dataset, data) {
   };
 }
 
-function doDelete() {
-  return function(res) {
-    return new Promise(function(resolve) {
-      $fh.sync.doDelete(datasetId, res.uid, function() {
-        resolve(res);
-      });
+function doDelete(datasetId, uid) {
+  return new Promise(function(resolve, reject) {
```

Comment:
I'm wondering whether in places like this we can take advantage of `Promise.promisify` to avoid the anti-pattern described here http://bluebirdjs.com/docs/anti-patterns.html#the-explicit-construction-anti-pattern ? We can take a look at implementing the change in a separate PR, wdyt?